REST. Boby "private int num;" llegalArgumentException: Can not set int field

Добрый день!
Есть необходимость в ресте отправлять значение num без кавычек.
Судя по документации необходимо в классе вместо String указать int. Но при указании int возникает ошибка.
Версии 
<pf2.version>14.9.0</pf2.version>
<pf2.version>13.0.3</pf2.version>

Класс rest.endpoints.Document
@Body(name = "num")
private int num;

 

ru.sbtqa.tag.api.exception.RestPluginException: Body with name "num" is not available

Caused by: java.lang.IllegalArgumentException: Can not set int field 
путь до класса.Document.num to java.lang.String

 

ru.sbtqa.tag.api.exception.RestPluginException: Body with name "num" is not available

                at ru.sbtqa.tag.api.utils.ReflectionUtils.set(ReflectionUtils.java:30)

                at ru.sbtqa.tag.api.EndpointEntryReflection.setParameterValueByTitle(EndpointEntryReflection.java:93)

                at ru.sbtqa.tag.api.EndpointEntry.send(EndpointEntry.java:60)

                at ru.sbtqa.tag.api.junit.ApiStepsImpl.send_aroundBody6(ApiStepsImpl.java:79)

                at ru.sbtqa.tag.api.junit.ApiStepsImpl$AjcClosure7.run(ApiStepsImpl.java:1)

                at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)

                at ru.sbtqa.tag.pagefactory.allure.JunitReporter.handleStep(JunitReporter.java:34)

                at ru.sbtqa.tag.api.aspects.JUnitApiStepAspect.handleStep(JUnitApiStepAspect.java:13)

                at ru.sbtqa.tag.api.junit.ApiStepsImpl.send(ApiStepsImpl.java:78)

                at ru.sbtqa.tag.stepdefs.ru.ApiStepDefs.send(ApiStepDefs.java:38)

                at ✽.он отправляет запрос "Document" с параметрами:(src/test/resources/features/rest/test1.feature:7)

Caused by: java.lang.IllegalArgumentException: Can not set int field 
путь до класса.Document.num to java.lang.String

                at sun.reflect.UnsafeFieldAccessorImpl.throwSetIllegalArgumentException(UnsafeFieldAccessorImpl.java:167)

                at sun.reflect.UnsafeFieldAccessorImpl.throwSetIllegalArgumentException(UnsafeFieldAccessorImpl.java:171)

                at sun.reflect.UnsafeIntegerFieldAccessorImpl.set(UnsafeIntegerFieldAccessorImpl.java:98)

                at java.lang.reflect.Field.set(Field.java:764)

                at ru.sbtqa.tag.api.utils.ReflectionUtils.set(ReflectionUtils.java:28)

                at ru.sbtqa.tag.api.EndpointEntryReflection.setParameterValueByTitle(EndpointEntryReflection.java:93)

                at ru.sbtqa.tag.api.EndpointEntry.send(EndpointEntry.java:60)

                at ru.sbtqa.tag.api.junit.ApiStepsImpl.send_aroundBody6(ApiStepsImpl.java:79)

                at ru.sbtqa.tag.api.junit.ApiStepsImpl$AjcClosure7.run(ApiStepsImpl.java:1)

                at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)

                at ru.sbtqa.tag.pagefactory.allure.JunitReporter.handleStep(JunitReporter.java:34)

                at ru.sbtqa.tag.api.aspects.JUnitApiStepAspect.handleStep(JUnitApiStepAspect.java:13)

                at ru.sbtqa.tag.api.junit.ApiStepsImpl.send(ApiStepsImpl.java:78)

                at ru.sbtqa.tag.stepdefs.ru.ApiStepDefs.send(ApiStepDefs.java:38)

                at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)

                at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)

                at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)

                at java.lang.reflect.Method.invoke(Method.java:498)

                at cucumber.runtime.Utils$1.call(Utils.java:32)

                at cucumber.runtime.Timeout.timeout(Timeout.java:16)

                at cucumber.runtime.Utils.invoke(Utils.java:26)

                at cucumber.runtime.java.JavaStepDefinition.execute(JavaStepDefinition.java:37)

                at cucumber.runtime.StepDefinitionMatch.runStep_aroundBody0(StepDefinitionMatch.java:40)

                at cucumber.runtime.StepDefinitionMatch$AjcClosure1.run(StepDefinitionMatch.java:1)

                at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)

                at ru.sbtqa.tag.pagefactory.aspects.CriticalStepCheckAspect.runStep(CriticalStepCheckAspect.java:68)

                at cucumber.runtime.StepDefinitionMatch.runStep(StepDefinitionMatch.java:40)

                at cucumber.api.TestStep.executeStep_aroundBody2(TestStep.java:102)

                at cucumber.api.TestStep$AjcClosure3.run(TestStep.java:1)

                at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)

                at ru.sbtqa.tag.pagefactory.aspects.DataAspect.executeStep(DataAspect.java:106)

                at cucumber.api.TestStep.executeStep(TestStep.java:101)

                at cucumber.api.TestStep.run_aroundBody0(TestStep.java:83)

                at cucumber.api.TestStep$AjcClosure1.run(TestStep.java:1)

                at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)

                at ru.sbtqa.tag.pagefactory.aspects.DataAspect.run(DataAspect.java:87)

                at cucumber.api.TestStep.run(TestStep.java:78)

                at cucumber.api.TestCase.run(TestCase.java:58)

                at cucumber.runner.Runner.runPickle(Runner.java:80)

                at cucumber.runtime.junit.PickleRunners$NoStepDescriptions.run(PickleRunners.java:140)

                at cucumber.runtime.junit.FeatureRunner.runChild(FeatureRunner.java:68)

                at cucumber.runtime.junit.FeatureRunner.runChild(FeatureRunner.java:23)

                at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)

                at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)

                at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)

                at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)

                at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)

                at org.junit.runners.ParentRunner.run(ParentRunner.java:363)

                at cucumber.runtime.junit.FeatureRunner.run(FeatureRunner.java:73)

                at cucumber.api.junit.Cucumber.runChild(Cucumber.java:118)

                at cucumber.api.junit.Cucumber.runChild(Cucumber.java:56)

                at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)

                at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)

                at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)

                at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)

                at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)

                at cucumber.api.junit.Cucumber$1.evaluate(Cucumber.java:127)

                at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)

                at org.junit.runners.ParentRunner.run(ParentRunner.java:363)

                at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)

                at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:272)

                at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:236)

                at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)

                at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)

                at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)

                at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)

--- a/plugins/rest-plugin/pom.xml
+++ b/plugins/rest-plugin/pom.xml
@@ -33,6 +33,11 @@
       <artifactId>rest-assured</artifactId>
       <version>3.3.0</version>
     </dependency>
+    <dependency>
+      <groupId>commons-beanutils</groupId>
+      <artifactId>commons-beanutils</artifactId>
+      <version>1.9.4</version>
+    </dependency>
 
     <!--TESTS-->
     <dependency>

--- a/plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/junit/ApiStepsImpl.java
+++ b/plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/junit/ApiStepsImpl.java
@@ -175,7 +175,7 @@ public class ApiStepsImpl<T extends ApiStepsImpl<T>> {
      * @param value with this value the parameter will be added to endpoint
      * blank
      */
-    public T add(ParameterType type, String name, String value) {
+    public T add(ParameterType type, String name, Object value) {
         ApiEnvironment.getBlankStorage().getLast().addParameter(type, name, value);
         return (T) this;
     }

--- a/plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/storage/EndpointBlank.java
+++ b/plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/storage/EndpointBlank.java
@@ -1,9 +1,10 @@
 package ru.sbtqa.tag.api.storage;
 
+import ru.sbtqa.tag.api.annotation.ParameterType;
+import ru.sbtqa.tag.pagefactory.annotations.rest.Endpoint;
+
 import java.util.HashMap;
 import java.util.Map;
-import ru.sbtqa.tag.pagefactory.annotations.rest.Endpoint;
-import ru.sbtqa.tag.api.annotation.ParameterType;
 
 /**
  * The blank of the request.
@@ -31,7 +32,9 @@ public class EndpointBlank {
         return title;
     }
 
-    public void addParameter(ParameterType type, String name, String value) {
+    public void addParameter(ParameterType type, String name, Object valueObj) {
+        String value = String.valueOf(valueObj);
+
         if (value != null && !value.equals(""))
             switch (type) {
                 case QUERY:

--- a/plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/utils/ReflectionUtils.java
+++ b/plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/utils/ReflectionUtils.java
@@ -1,5 +1,6 @@
 package ru.sbtqa.tag.api.utils;
 
+import org.apache.commons.beanutils.ConvertUtils;
 import ru.sbtqa.tag.api.EndpointEntry;
 import ru.sbtqa.tag.api.annotation.Validation;
 import ru.sbtqa.tag.api.exception.RestPluginException;
@@ -26,8 +27,9 @@ public class ReflectionUtils {
 
     public static void set(EndpointEntry endpoint, Field field, Object value) {
         try {
+            Object convertedValue = ConvertUtils.convert(value, field.getType());
             field.setAccessible(true);
-            field.set(endpoint, value);
+            field.set(endpoint, convertedValue);
         } catch (IllegalArgumentException | IllegalAccessException ex) {
             throw new RestPluginException(format("Body with name \"%s\" is not available", field.getName()), ex);
         }

--- a/plugins/rest-plugin/src/test/java/ru/sbtqa/tag/api/JunitTests.java
+++ b/plugins/rest-plugin/src/test/java/ru/sbtqa/tag/api/JunitTests.java
@@ -81,7 +81,7 @@ public class JunitTests {
                 .add(ParameterType.HEADER, headers)
                 .add(ParameterType.QUERY, "query-parameter-name-1", "query-parameter-value-1")
                 .add(ParameterType.QUERY, queries)
-                .add(ParameterType.BODY, "id", "11223344")
+                .add(ParameterType.BODY, "id", 11223344)
                 .add(ParameterType.BODY, bodies)
                 .send().validate();
     }

--- a/plugins/rest-plugin/src/test/java/ru/sbtqa/tag/api/endpoints/ClientJsonEndpoint.java
+++ b/plugins/rest-plugin/src/test/java/ru/sbtqa/tag/api/endpoints/ClientJsonEndpoint.java
@@ -22,6 +22,10 @@ import ru.sbtqa.tag.api.dto.SimpleResult;
 import ru.sbtqa.tag.api.utils.Default;
 import ru.sbtqa.tag.api.utils.TestDataUtils;
 
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+
 @Path("/client")
 public class ClientJsonEndpoint {
 
@@ -179,15 +183,21 @@ public class ClientJsonEndpoint {
     @Produces(MediaType.APPLICATION_JSON)
     public Response replace(String days) {
 
-        BasicDBObject expected = (BasicDBObject) JSON.parse("{ \"day1\" :  null  , \"day11\" : \"null\" , " +
-                "\"day13\" : \"true\" , \"day14\" : \"true\" , \"day15\" : \"\"}");
+        BasicDBObject expected = (BasicDBObject) JSON.parse("" +
+                "{\n" +
+                "  \"day1\": false,\n" +
+                "  \"day11\": false,\n" +
+                "  \"day12\": false,\n" +
+                "  \"day13\": true,\n" +
+                "  \"day14\": true,\n" +
+                "  \"day15\": false\n" +
+                "}");
 
         BasicDBObject actual = (BasicDBObject) JSON.parse(days);
 
-        SimpleResult result = new SimpleResult();
-        result.setResult(String.valueOf(expected.equals(actual)));
+        BasicDBObject result = new BasicDBObject();
+        result.put("result", expected.equals(actual));
 
-        return Response.ok(result)
-                .build();
+        return Response.ok(result.toJson()).build();
     }
 }

--- a/plugins/rest-plugin/src/test/java/ru/sbtqa/tag/api/entries/template/ReplacerEntry.java
+++ b/plugins/rest-plugin/src/test/java/ru/sbtqa/tag/api/entries/template/ReplacerEntry.java
@@ -15,25 +15,25 @@ public class ReplacerEntry extends EndpointEntry {
     private String header = "application/json";
 
     @Body(name = "day1")
-    private String day1 = null;
+    private Boolean day1 = null;
 
     @Body(name = "day11")
-    private String day11 = "true";
+    private boolean day11 = true;
 
     @Body(name = "day12")
-    private String day12;
+    private boolean day12;
 
     @Body(name = "day13")
-    private String day13 = "true";
+    private boolean day13 = true;
 
     @Body(name = "day14")
-    private String day14 = "true";
+    private boolean day14 = true;
 
     @Body(name = "day15")
-    private String day15;
+    private boolean day15 = true;
 
     @Validation(title = "works correctly")
     public void validate() {
-        getResponse().body("result", equalTo("true"));
+        getResponse().body("result", equalTo(true));
     }
 }

--- a/plugins/rest-plugin/src/test/resources/data/Positive.json
+++ b/plugins/rest-plugin/src/test/resources/data/Positive.json
@@ -1,1 +1,1 @@
-{  "Vehicle": {    "model": "3er",    "brand": "BMW",    "year": "2014"  }}
+{  "Vehicle": {    "model": "3er",    "brand": "BMW",    "year": 2014  }}

